hourly-tips/src/main/java/org/apache/flink/training/exercises/hourlytips/HourlyTipsExercise.java

```diff
@@ -40,18 +56,32 @@
      */
     public static void main(String[] args) throws Exception {
 
+        HourlyTipsSolution job =
+                new HourlyTipsSolution(new TaxiFareGenerator(), new PrintSinkFunction<>());
+
+        job.execute();
+    }
+
+    /**
+     * Create and execute the hourly tips pipeline.
+     *
+     * @return {JobExecutionResult}
+     * @throws Exception which occurs during job execution.
+     */
+    public JobExecutionResult execute() throws Exception {
+
         // set up streaming execution environment
         StreamExecutionEnvironment env = StreamExecutionEnvironment.getExecutionEnvironment();
-        env.setParallelism(ExerciseBase.parallelism);
 
         // start the data generator
-        DataStream<TaxiFare> fares = env.addSource(fareSourceOrTest(new TaxiFareGenerator()));
-
-        throw new MissingSolutionException();
+        DataStream<TaxiFare> fares = env.addSource(new TaxiFareGenerator());
```

Comment:
```suggestion
        DataStream<TaxiFare> fares = env.addSource(source);
```

Reply:
Similarly, we should hint users to use `sink` in their solutions or the tests will fail